autoload/sj/elixir.vim

```diff
@@ -189,3 +189,85 @@ function! sj#elixir#JoinArray()
 
   return 1
 endfunction
+
+function! sj#elixir#SplitPipe()
+  let line = getline('.')
+
+  if line =~ '^\s*|>\s\+'
+    return 0
+  endif
+
+  call sj#PushCursor()
+  normal! 0f(
```

Comment:
This is tricky. Since a function doesn't *have* to have round brackets, this could jump to the wrong place. Consider this:

``` elixir
foo "one" + bar("two")
```

I kind of understand why you go to the start of the line. In a `foo("one") + bar("two")` situation, you don't want to pull out the "two" of `bar`, since it would be syntactically incorrect -- you need the leftmost function call. But in this case, you won't get it, since jumping with `f(` puts you on the `bar(`:

``` elixir
"two"
|> foo "one" + bar()
```

Maybe something like this instead, to jump to the end of the closest function-looking thing after the start of the line?

```suggestion
  normal! 0
  if search('\k\+[!?]\=[ (].', 'We', line('.')) <= 0
    return 0
  endif
```

The search function can be difficult to read, this particular combination of flags makes it jump to the end of the pattern, limited to the current line: [`:help search()`](https://vimhelp.org/eval.txt.html#search%28%29)

Reply:
As discussed above, I think with this example there's no a good way to split it into a pipeline (with or without parens on the function call). So I'd rather just find a way to ignore it altogether and focus only on calls that occupy the entire line, if you're happy with that?

Reply:
Sure, that makes sense -- I would consider comments, like `foo("bar") # baz`, but the comment being removed might be fine.

Reply:
Well, in this case, `foo "one" + bar("two")`, the single argument to `foo` would be `"one" + bar("two")`, but I suppose transforming it into `"one" + bar("two") |> foo` would be incorrect without round brackets around the pre-pipe expression.